tests/PHPStan/Rules/Functions/data/bug-9803.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bug9803;
+
+function doFoo() {
+	$random = rand(1, 5);
+	$array = array("one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten");
+
+	$keys = array();
+	if ($random == 1)
+		$keys = array(array_rand($array));
+	else
+		$keys = array_rand($array, $random);
```

Comment:
Would be nice to sprinkle some assertType calls here, and reference it from NodeScopeResolverTest.

And please add `{ ... }` around those `if` and `else`. Thanks!

Reply:
done